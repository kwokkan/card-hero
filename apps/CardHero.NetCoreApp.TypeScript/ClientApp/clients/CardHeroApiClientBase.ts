// no export as this is used by NSwag generated api clients
class CardHeroApiClientBase {
    getBaseUrl(defaultBaseUrl: string, baseUrl: string): string {
        if (!baseUrl) {
            return defaultBaseUrl;
        }

        if (baseUrl.endsWith('/')) {
            // remove trailing slash as NSwag uses '/' as part of the api url
            return baseUrl.substring(0, baseUrl.length - 2);
        }

        return baseUrl;
    }
}

module.exports = CardHeroApiClientBase;