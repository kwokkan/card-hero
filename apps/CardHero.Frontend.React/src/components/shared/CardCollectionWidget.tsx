import React from "react";
import { ICardCollectionModel } from "../../clients/clients";
import { CardCollectionCard } from "./CardCollectionCard";

interface ICardCollectionWidgetProps {
    title?: string;
    cardCollection: ICardCollectionModel[];
    cardActionName?: string;
    onCardClicked?: (card: ICardCollectionModel) => void;
    cardActionDisabled?: boolean;
    cardActionClassName?: string;
    subSection?: JSX.Element;
}

export function CardCollectionWidget(props: ICardCollectionWidgetProps) {
    const cardCollection = props.cardCollection || [];

    return (
        <div className="card">
            {props.title &&
                <h4 className="card-header">
                    {props.title}
                </h4>
            }

            {props.subSection &&
                <div className="card-body">
                    {props.subSection}
                </div>
            }

            <div className="list-group list-group-flush">
                {cardCollection.length > 0 ?
                    (cardCollection.map(cc =>
                        <CardCollectionCard
                            key={cc.id}
                            card={cc}
                            actionName={props.cardActionName}
                            onActionClicked={props.onCardClicked}
                            actionDisabled={props.cardActionDisabled}
                            actionClassName={props.cardActionClassName}
                        />
                    ))
                    :
                    (
                        <div className="list-group-item">
                            No cards
                        </div>
                    )
                }
            </div>
        </div>
    );
}
