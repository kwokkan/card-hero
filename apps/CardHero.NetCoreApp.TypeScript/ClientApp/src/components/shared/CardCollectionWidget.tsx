import React from "react";
import { ICardCollectionModel } from "../../clients/clients";

interface ICardCollectionWidgetProps {
    title?: string;
    cardCollection: ICardCollectionModel[];
}

export function CardCollectionWidget(props: ICardCollectionWidgetProps) {
    const cardCollection = props.cardCollection || [];

    return (
        <div className="card">
            <div className="card-body">
                {props.title &&
                    <h4 className="card-title">
                        {props.title}
                    </h4>
                }
                <div className="card-text">
                    <ul className="ch-cards droppable">
                        {cardCollection.map(cc =>
                            <li key={cc.id} className="ch-card draggable" data-card-collection-id={cc.id}>{cc.card.name}</li>
                        )}
                    </ul>
                </div>
            </div>
        </div>
    );
}
