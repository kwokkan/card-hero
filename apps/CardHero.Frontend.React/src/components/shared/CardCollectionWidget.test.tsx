import React from "react";
import renderer from "react-test-renderer";
import { CardModel, ICardCollectionModel, ICardModel } from "../../clients/clients";
import { CardCollectionWidget } from "./CardCollectionWidget";

test("<CardCollectionWidget /> renders no cards", () => {
    const tree = renderer
        .create(<CardCollectionWidget cardCollection={[]} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<CardCollectionWidget /> renders card with custom actionName", () => {
    const tree = renderer
        .create(<CardCollectionWidget cardCollection={[]} title="Test widget" />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<CardCollectionWidget /> renders title with no cards", () => {
    const tree = renderer
        .create(<CardCollectionWidget cardCollection={[]} title="Test title" />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<CardCollectionWidget /> renders card with cards", () => {
    const data1: ICardModel = {
        id: 1,
        name: "Test card"
    };
    const data2: ICardModel = {
        id: 2,
        name: "Test card 2"
    };
    const card: ICardCollectionModel[] = [
        {
            card: CardModel.fromJS(data1)
        }, {
            card: CardModel.fromJS(data2)
        }
    ];
    const tree = renderer
        .create(<CardCollectionWidget cardCollection={card} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<CardCollectionWidget /> renders card with cards and custom actions", () => {
    const data1: ICardModel = {
        id: 1,
        name: "Test card"
    };
    const data2: ICardModel = {
        id: 2,
        name: "Test card 2"
    };
    const card: ICardCollectionModel[] = [
        {
            card: CardModel.fromJS(data1)
        }, {
            card: CardModel.fromJS(data2)
        }
    ];
    const tree = renderer
        .create(<CardCollectionWidget cardCollection={card} cardActionName="Click me" cardActionClassName="my-class" />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<CardCollectionWidget /> renders card with cards disabled", () => {
    const data1: ICardModel = {
        id: 1,
        name: "Test card"
    };
    const data2: ICardModel = {
        id: 2,
        name: "Test card 2"
    };
    const card: ICardCollectionModel[] = [
        {
            card: CardModel.fromJS(data1)
        }, {
            card: CardModel.fromJS(data2)
        }
    ];
    const tree = renderer
        .create(<CardCollectionWidget cardCollection={card} cardActionName="Click me" cardActionDisabled={true} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});
