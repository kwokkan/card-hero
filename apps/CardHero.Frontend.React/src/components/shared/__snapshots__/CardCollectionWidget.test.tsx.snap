// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCollectionWidget /> renders card with cards 1`] = `
<div
  className="card"
>
  <div
    className="list-group list-group-flush"
  >
    <div
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      Test card
    </div>
    <div
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      Test card 2
    </div>
  </div>
</div>
`;

exports[`<CardCollectionWidget /> renders card with cards and custom actions 1`] = `
<div
  className="card"
>
  <div
    className="list-group list-group-flush"
  >
    <div
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      Test card
      <button
        className="btn btn-sm my-class"
        onClick={[Function]}
        type="button"
      >
        Click me
      </button>
    </div>
    <div
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      Test card 2
      <button
        className="btn btn-sm my-class"
        onClick={[Function]}
        type="button"
      >
        Click me
      </button>
    </div>
  </div>
</div>
`;

exports[`<CardCollectionWidget /> renders card with cards disabled 1`] = `
<div
  className="card"
>
  <div
    className="list-group list-group-flush"
  >
    <div
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      Test card
      <button
        className="btn btn-sm "
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Click me
      </button>
    </div>
    <div
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      Test card 2
      <button
        className="btn btn-sm "
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Click me
      </button>
    </div>
  </div>
</div>
`;

exports[`<CardCollectionWidget /> renders card with custom actionName 1`] = `
<div
  className="card"
>
  <h4
    className="card-header"
  >
    Test widget
  </h4>
  <div
    className="list-group list-group-flush"
  >
    <div
      className="list-group-item"
    >
      No cards
    </div>
  </div>
</div>
`;

exports[`<CardCollectionWidget /> renders no cards 1`] = `
<div
  className="card"
>
  <div
    className="list-group list-group-flush"
  >
    <div
      className="list-group-item"
    >
      No cards
    </div>
  </div>
</div>
`;

exports[`<CardCollectionWidget /> renders title with no cards 1`] = `
<div
  className="card"
>
  <h4
    className="card-header"
  >
    Test title
  </h4>
  <div
    className="list-group list-group-flush"
  >
    <div
      className="list-group-item"
    >
      No cards
    </div>
  </div>
</div>
`;
